internal/repository/repack.go

```diff
@@ -80,7 +77,7 @@ func Repack(ctx context.Context, repo restic.Repository, packs restic.IDSet, kee
 			if !id.Equal(entry.ID) {
 				debug.Log("read blob %v/%v from %v: wrong data returned, hash is %v",
 					h.Type, h.ID, tempfile.Name(), id)
-				fmt.Fprintf(os.Stderr, "read blob %v from %v: wrong data returned, hash is %v",
+				return nil, errors.Errorf("read blob %v from %v: wrong data returned, hash is %v",
```

Comment:
I see that there are error cases (like a magnetic hard disk starting to fail), where you would like to immediately stop at this point and start a forensic analysis and/or try to move all of you repository as fast as possible to some save place.

But, again, we leave the repository in a state where some new pack files + index files might have been written. This could cause problems like full disks or memory issues. Actually I think in the long run we need some "cleanup" within `prune` that tries to remove the newly generated repo files in case of an error.

For now, I agree that this improves safety.

Reply:
Hard disk errors would already be caught by the `repo.Key().Open` call a few lines earlier. This check verifies that the blob we've just read is actually the expected one. There are two possible causes for this: Either the blob was damaged before encrypting (e.g. a bit-flip in memory) or someone replaced the encrypted blob bytes with the bytes of another encrypted blob of the same size. Both scenarios indicate a potentially catastrophic repository damage.

The current error handling strategy in prune is to just stop whenever some repository damage is encountered. Changing this strategy is completely out of scope for this PR.

The full disk problematic is not affected by this error check. The other error checks in will cause `repack` to abort when it encounters a damaged blob.

Reply:
:+1: